### PR TITLE
Use go template for flannel operator

### PR DESCRIPTION
**Pull request by MarcelMue**
towards https://github.com/giantswarm/giantswarm/issues/4388

### CODE REVIEW

File: integration/setup/key.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// +build k8srequired
+
+package setup
+
+import (
+	"fmt"
+)
+
+func clusterRole(clusterID, operator string) string {
+	return fmt.Sprintf("%s-%s", clusterID, operator)
```
Comment (xh3b4sd) [inline integration/setup/key.go]: I think we should import the env package here and use the tenant ID implicitly.

File: integration/setup/key.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// +build k8srequired
+
+package setup
```
Comment (xh3b4sd) [inline integration/setup/key.go]: I think this should be a separate `key` package.

**Review by xh3b4sd — APPROVED** [review]
That is getting places here.

File: integration/setup/key.go (hunk 2 above)
Comment (MarcelMue) [reply integration/setup/key.go]: Sorted. I will improve the `env` stuff in a future PR. It is a bit all over the place currently.

**Review by calvix — APPROVED** [review]
lgtmifwy

**Review by kopiczko — APPROVED** [review]

File: integration/key/key.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	return fmt.Sprintf("%s-%s", env.ClusterID(), operator)
+}
+
+func ClusterRolePSP(operator string) string {
+	return fmt.Sprintf("%s-%s-psp", env.ClusterID(), operator)
+}
+
+func PSPName(operator string) string {
+	return fmt.Sprintf("%s-%s", env.ClusterID(), operator)
+}
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply integration/key/key.go]: I'll align this whole foo in a later PR.

**Pull request merged**